doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+
+        Despite this informal contract with our users, the software is provided
+        as is, as stated in the licence.  When a release inadvertently
+        introduces changes that are not backwards compatible, these are known
+        as software regressions.
+
+    categorical feature
+        A categorical or nominal :term:`feature` is one that has a finite set
+        of discrete values across the population of data.  These should be
+        represented as a column of integers. In representation, categorical
+        features are not distinguished from other integer features: ordinal or
+        count-valued. In the future we hope to support string-valued
+        categorical features. Some estimators may handle categorical features
+        better when one-hot encoded.  See the
+        `http://contrib.scikit-learn.org/categorical-encoding
+        <category_encoders>`_ package,
+        :class:`~sklearn.preprocessing.OneHotEncoder`, and a proposed
+        :issue:`9151 <CategoricalEncoder>` for tools related to encoding
+        categorical features.
+
+    clone
+        To copy an :term:`estimator instance` and create a new one with
+        identical :term:`parameters`, but without any fitted
+        :term:`attributes`, using :func:`~skelarn.base.clone`.
+
+        When ``fit`` is called, a :term:`meta-estimator` usually clones
+        a wrapped estimator instance before fitting the cloned instance.
+        (Exceptions, for legacy reasons, include
+        :class:`~pipeline.Pipeline` and
+        :class:`~pipeline.FeatureUnion`.)
+
+    common tests
+        This refers to the tests run on almost every estimator class in
+        Scikit-learn to check they comply with basic API conventions.  They are
+        available for external use through
+        :func:`utils.estimator_checks.check_estimator`.
+
+        Note: Some exceptions to the common testing regime are currently
+        hard-coded into the library, but we hope to replace this by marking
+        exceptional behaviours on the estimator using semantic :term:`estimator
+        tags`.
+
+    deprecation
+        We use deprecation to slowly violate our :term:`backwards
+        compatibility` assurances, usually to to:
+
+        * change the semantics of the default value of a parameter; or
+        * remove a parameter, attribute, method, class, etc.
+
+        We will ordinarily issue a warning when a deprecated element is used,
+        although there may be limitations to this.  For instance, we will raise
+        a warning when someone sets a parameter that has been deprecated, but
+        may not when they access that parameter's attribute on the estimator
+        instance.
+
+        See the :ref:`Contributors' Guide <contributing_deprecation>`.
+
+    docstring
+        TODO
+
+    double underscore notation
+        When specifying parameter names for nested estimators, ``__`` may be
+        used to separate between parent and child.
```

Comment:
in ``set_params``, ``get_params`` and ``_search.py``? That's the only context, right?

Reply:
Also pipeline's fit_params....